docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,328 @@
+- The :class:`~chainer.dataset.Iterator` 's constructor takes two arguments: a dataset object and a mini-batch size.
+- If you want to use the same dataset repeatedly during the training process, set the ``repeat`` argument to ``True`` (default). Otherwise, the dataset will be used only one time. The latter case is actually for the evaluation.
+- If you want to shuffle the training dataset every epoch, set the ``shuffle`` argument to ``True``. Otherwise, the order of each data retrieved from the dataset will be always the same at each epoch.
+
+In the example code shown above, we set ``batchsize = 128`` create both ``train_iter`` and ``test_iter``. So, these iterators will provide 128 images and corresponding labels at a time.
+
+3. Define a network
+'''''''''''''''''''
+
+Now let's define a neural network that we will train to classify the MNIST images. For simplicity, we use a three-layer perceptron here. We set each hidden layer to have 100 units and set the output layer to have 10 units, which is corresponding to the number of class labels of the MNIST.
+
+Create your network as a subclass of Chain
+..........................................
+
+You can create your network by writing a new subclass of :class:`~chainer.Chain`.
+The main steps are twofold:
+
+1. Register the network components which have trainable parameters to the subclass. Each of them must be instantiated and assigned to a property in the scope specified by :meth:`~chainer.Chain.init_scope`:
+
+2. Define a :meth:`~chainer.Chain.__call__` method that represents the actual **forward computation** of your network. This method takes one or more :class:`~chainer.Variable`, :class:`numpy.array`, or :class:`cupy.array` as its inputs and calculate the forward pass using them.
+
+    .. testcode::
+
+        class MyNetwork(Chain):
+
+            def __init__(self, n_mid_units=100, n_out=10):
+                super().__init__()
+                with self.init_scope():
+                    self.l1 = Linear(None, n_mid_units)
+                    self.l2 = Linear(n_mid_units, n_mid_units)
+                    self.l3 = Linear(n_mid_units, n_out)
+
+            def __call__(self, x):
+                h = F.relu(self.l1(x))
+                h = F.relu(self.l2(h))
+                return self.l3(h)
+
+It should be noted that only :class:`~chainer.Link`, :class:`~chainer.Chain`, and :class:`~chainer.ChainList` objects can be registered to the model by assinging it as a property inside the :meth:`~chainer.Chain.init_scope`. This is because they can contain trainable parameters. For example, a :class:`~chainer.Function` does not contain any trainable parameters, so there is no need to keep the object as a property of your network. When you want to use :meth:`~chainer.functions.relu` in your network, just simply using it as a function in :meth:`~chainer.Chain.__call__` works correctly.
```

Comment:
+It should be noted that only :class:`~chainer.Link`, :class:`~chainer.Chain`, and :class:`~chainer.ChainList` objects can be registered to the model by assigning it as a property inside the :meth:`~chainer.Chain.init_scope`. This is because they can contain trainable parameters. For example, a :class:`~chainer.Function` does not contain any trainable parameters, so there is no need to keep the object as a property of your network. When you want to use :meth:`~chainer.functions.relu` in your network, using it as a function in :meth:`~chainer.Chain.__call__` works correctly.